src/libs/ajax/fetchAdapter.js

```diff
@@ -0,0 +1,68 @@
+import { redirectOnLogout, reportError } from 'src/libs/ajax.js'
+
+function buildUrlWithParams(url, params = {}) {
+  const query = new URLSearchParams(params).toString()
+  return query ? `${url}?${query}` : url
+}
+
+async function handleResponse(res, url, responseType, method = 'GET') {
+  const isMeCheck = url?.endsWith('/api/user/me') && (method?.toLowerCase() === 'get')
+  if (!res.ok) {
+    if (res.status === 401 && !isMeCheck) {
+      redirectOnLogout()
+    }
+    if (res.status >= 500) {
+      await reportError(url, res.status)
```

Comment:
If a function only does reporting, logging, analytics, or metrics, then blocking execution flow on the return of that function is generally not needed.  The cost is unnecessary slowness -- potentially quite a bit of slowness if the logging service is having issues -- and there's typically no benefit as the function doesn't return any value.  So omitting `await` in such cases helps speed and graceful degradation.


```suggestion
      // Don't `await`, as we don't need the result
      reportError(url, res.status)
```